File: 2020/ruby/spec/day01_spec.rb
require 'rspec'
require_relative '../lib/day01'

describe ExpenseAudit do
  describe "part1 - example" do
    it "finds the pair that sums to 2020" do
      audit = described_class.new([1721, 979, 366, 299, 675, 1456])
      result = audit.which_pair_is_2020
      expect(result).to eq([1721, 299])
    end
  end

  describe "part 2 - example" do
    it "finds the triplet that sums to 2020" do
      audit = described_class.new([1721, 979, 366, 299, 675, 1456])
      result = audit.which_triplet_is_2020
      expect(result).to eq([979, 366, 675]) 
    end
  end
end

File: 2020/node/src/__tests__/day01.spec.ts
import { which_pair_is_2020 } from '../day01';

const example_expenses: number[] = [1721, 979, 366, 299, 675, 1456]

test('which_pair_is_2020', () => {
  expect(which_pair_is_2020(example_expenses)).toEqual([1721, 299]);
});